relay-metrics/src/aggregation.rs

```diff
@@ -520,6 +520,21 @@ impl BucketValue {
             Self::Distribution(m) => m.internal_size(),
         }
     }
+
+    /// Estimates the number of bytes needed to encode the bucket.
+    /// Note that this does not necessarily match the exact memory footprint of the bucket,
+    /// because datastructures might have a memory overhead.
+    ///
+    /// This is very similar to [`BucketValue::relative_size`], which can possibly be removed.
+    pub fn cost(&self) -> usize {
+        match self {
+            Self::Counter(_) => 8,
```

Comment:
Instead of hard-coding, use `std::mem::size_of` with typedefs for the values. 

Also, note that this is wrong. The size of a bucket value is always `size_of::<BucketValue>` + the allocations that happen within. So a more correct implementation would be:

```rust
const DIST_SIZE: usize = mem::size_of::<f64>() + mem::size_of::<Count>();

let allocations = match self {
    Self::Counter(_) => 0,
    Self::Set(s) => s.len() * mem::size_of::<u32>(), // better to typedef this to `SetValue` now
    Self::Gauge(_) => 0,
    Self::Distribution(m) => m.internal_size() * DIST_SIZE,
};

mem::size_of::<Self>() + allocations
```

Reply:
I had an implementation that used `size_of` (at least partially), but @untitaker argued that having an explicit model that has to be changed manually is better: https://github.com/getsentry/relay/pull/1284#discussion_r887927622

Reply:
This is still hared-coded and explicit, it just doesn't use magic numbers to be more self-explanatory.